File: 프로그래머스/0/120888. 중복된 문자 제거/중복된 문자 제거.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string my_string) {
    // visited 이용 방법
    
//     string answer = "";
//     bool visited[128] = {false};
    
//     for(char c : my_string) {
//         if(!visited[c]) {
//             answer += c;
//             visited[c] = true;
//         }
//     }
    
    // string.find 이용 방법 (추천)
    string answer = "";
    for(char c : my_string) {
        if(answer.find(c) == string::npos) {
            answer += c;
        }
    }
    return answer;
}

File: 프로그래머스/0/120888. 중복된 문자 제거/중복된 문자 제거.swift
import Foundation

extension Array where Element: Equatable {
    var unique: [Element] {
        var uniqueValues: [Element] = []
        forEach { item in
            guard !uniqueValues.contains(item) else { return }
            uniqueValues.append(item)
        }
        return uniqueValues
    }
}

func solution(_ my_string:String) -> String {
    Array(my_string).unique.map { String($0) }.joined()
}
